rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -401,6 +333,12 @@ void PublisherData::copy_gid(uint8_t out_gid[RMW_GID_STORAGE_SIZE]) const
   entity_->copy_gid(out_gid);
 }
 
+std::vector<uint8_t> PublisherData::copy_gid() const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  return entity_->copy_gid();
+}
```

Comment:
I'm not a fan of having two overloads for `copy_gid` here.  Can we reduce this to one?